addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -678,13 +678,13 @@ def write(self, vals):
 
         for applicant in self:
             if applicant.pool_applicant_id and (not applicant.is_pool_applicant):
```

Comment:
```suggestion
            if applicant.pool_applicant_id and applicant.id != applicant.pool_applicant_id and (not applicant.is_pool_applicant):
```
Issue was generated because we got applicant and applicant.pool_applicant_id same record